test/algorithms/is_empty.cpp

```diff
@@ -0,0 +1,332 @@
+    tester::apply("MULTIPOINT(0 0,0 0,1 1)", false);
+}
+
+BOOST_AUTO_TEST_CASE( test_multilinestring )
+{
+    typedef test_is_empty<multi_linestring> tester;
+
+    tester::apply("MULTILINESTRING()", true);
+    tester::apply("MULTILINESTRING(())", true);
+    tester::apply("MULTILINESTRING((),())", true);
+    tester::apply("MULTILINESTRING((),(0 0))", false);
+    tester::apply("MULTILINESTRING((),(0 0),())", false);
+    tester::apply("MULTILINESTRING((0 0))", false);
+    tester::apply("MULTILINESTRING((0 0,1 0))", false);
+    tester::apply("MULTILINESTRING((),(),(0 0,1 0))", false);
+    tester::apply("MULTILINESTRING((0 0,1 0,0 1),(0 0,1 0,0 1,0 0))", false);
+}
+
+template <typename OpenRing>
+void test_open_ring()
+{
+    typedef test_is_empty<OpenRing> tester;
+
+    tester::apply("POLYGON(())", true);
+    tester::apply("POLYGON((0 0))", false);
+    tester::apply("POLYGON((0 0,1 0))", false);
+    tester::apply("POLYGON((0 0,1 0,0 1))", false);
+    tester::apply("POLYGON((0 0,0 0,1 0,0 1))", false);
+}
+
+template <typename ClosedRing>
+void test_closed_ring()
+{
+    typedef test_is_empty<ClosedRing> tester;
+
+    tester::apply("POLYGON(())", true);
+    tester::apply("POLYGON((0 0))", false);
+    tester::apply("POLYGON((0 0,0 0))", false);
+    tester::apply("POLYGON((0 0,1 0,0 0))", false);
+    tester::apply("POLYGON((0 0,1 0,0 1,0 0))", false);
+    tester::apply("POLYGON((0 0,1 0,1 0,0 1,0 0))", false);
+}
+
+BOOST_AUTO_TEST_CASE( test_ring )
+{
+    test_open_ring<ring_ccw_open>();
+    test_open_ring<ring_cw_open>();
+    test_closed_ring<ring_ccw_closed>();
+    test_closed_ring<ring_cw_closed>();
+}
+
+template <typename OpenPolygon>
+void test_open_polygon()
+{
+    typedef test_is_empty<OpenPolygon> tester;
+
+    tester::apply("POLYGON(())", true);
+    tester::apply("POLYGON((),())", true);
+    tester::apply("POLYGON((),(),())", true);
+    tester::apply("POLYGON((),(0 0,0 1,1 0))", true);
+    tester::apply("POLYGON((),(0 0,0 1,1 0),())", true);
+    tester::apply("POLYGON((),(),(0 0,0 1,1 0))", true);
+    tester::apply("POLYGON((0 0))", false);
+    tester::apply("POLYGON((0 0,10 0),(0 0))", false);
+    tester::apply("POLYGON((0 0,10 0),(1 1,2 1))", false);
+    tester::apply("POLYGON((0 0,10 0,0 10))", false);
+    tester::apply("POLYGON((0 0,10 0,0 10),())", false);
+    tester::apply("POLYGON((0 0,10 0,0 10),(1 1))", false);
+    tester::apply("POLYGON((0 0,10 0,0 10),(1 1,2 1))", false);
+    tester::apply("POLYGON((0 0,10 0,0 10),(1 1,2 1,1 2))", false);
+    tester::apply("POLYGON((0 0,10 0,10 10,0 10),(1 1,2 1,1 2))", false);
+}
+
+template <typename ClosedPolygon>
+void test_closed_polygon()
+{
+    typedef test_is_empty<ClosedPolygon> tester;
+
+    tester::apply("POLYGON(())", true);
+    tester::apply("POLYGON((),())", true);
+    tester::apply("POLYGON((),(),())", true);
+    tester::apply("POLYGON((),(0 0,0 1,1 0,0 0))", true);
+    tester::apply("POLYGON((),(0 0,0 1,1 0,0 0),())", true);
```

Comment:
Well, this is an artifact of the the Cartesian coordinate system. In spherical equatorial and geographic we ***need*** to consider interior rings of polygons in order to correctly compute MBRs.